images/e2e-dind-k3d/init.sh

```diff
@@ -4,9 +4,21 @@ set -e
 
 LOG_DIR=${ARTIFACTS:-"/var/log"}
 
-if [[ -n "${DOCKER_IN_DOCKER_ENABLED}" ]]; then
+if [[ "${DOCKER_IN_DOCKER_ENABLED}" == "true" ]]; then
   echo "[* * *] Starting Docker in Docker"
   dockerd --data-root=/docker-graph > "${LOG_DIR}/dockerd.log" 2>&1 &
 fi
 
+if [[ "$K3D_ENABLED" == "true" ]]; then
+  ARGS=()
+  echo -n "[ * * * ] Provisioning k3d cluster"
+  if [[ "$PROVISION_REGISTRY" == "true" ]]; then
+    echo " with registry k3d-registry.localhost:5000"
+    k3d registry create registry.localhost --port 5000
+    ARGS+=( "--registry-use=k3d-registry.localhost:5000" )
+  else
+    echo
```

Comment:
Do you need this empty else statement?

Reply:
Yes you do. At first `echo -n` will return a string without a new line. This logic is to add additional information based on the registry provisioning conditional. if-else is needed here to terminate the echo with new line